### PR TITLE
add multipart annotation

### LINKED ISSUE #2: Retrofit multipart form support
Hey @tejainece what the plan for this, of course I want to try the lib and first call I need to implement (login) is in form data lol 
Can you tell me the steps and I'll do a PR for this :) 

Do we go for a `@AsFormData` annotation ?

### PR DESCRIPTION
**Pull request by jaumard**
fix #2
@tejainece should be ok like this :)

### CODE REVIEW

**Comment by jaumard** [comment]
don't merge yet, I've get type `'List<Body>' is not a subtype of type 'MultipartForm' in type cast` from generator with your changes, need to check

**Comment by jaumard** [comment]
Ok all good now :)

File: retrofit_gen/lib/writer/writer.dart [hunk 1]
```diff
@@ -54,6 +54,14 @@ class Writer {
       if (body is FormFieldBody) {
         sb.write('.urlEncodedFormField(${body.key}, ${body.name})');
       }
+
+      if (body is MultipartForm) {
+        sb.write('.multipart((serializers.to(${body.name}) as Map<String, dynamic>).map((key, value) => MapEntry(key, value.toString())))');
```
Comment (tejainece) [inline retrofit_gen/lib/writer/writer.dart]: ` as Map<String, dynamic>).map((key, value) => MapEntry(key, value.toString())` is not needed. We can remove that.

**Comment by tejainece** [comment]
Looks good. 

If you remove `as Map<String, dynamic>).map((key, value) => MapEntry(key, value.toString()`, we can merge it.

File: retrofit_gen/lib/writer/writer.dart (hunk 1 above)
Comment (jaumard) [reply retrofit_gen/lib/writer/writer.dart]: I'll try but without it doesn't work ^^ let me try

**Comment by jaumard** [comment]
Just removed and still worked :) removed :)

**Pull request merged**